lib/taskana-core-test/src/test/java/acceptance/task/complete/CompleteTaskWithSpiAccTest.java

```diff
@@ -0,0 +1,144 @@
+import pro.taskana.testapi.TaskanaInject;
+import pro.taskana.testapi.TaskanaIntegrationTest;
+import pro.taskana.testapi.WithServiceProvider;
+import pro.taskana.testapi.builder.TaskBuilder;
+import pro.taskana.testapi.builder.WorkbasketAccessItemBuilder;
+import pro.taskana.testapi.security.WithAccessId;
+import pro.taskana.workbasket.api.WorkbasketPermission;
+import pro.taskana.workbasket.api.WorkbasketService;
+import pro.taskana.workbasket.api.models.WorkbasketSummary;
+
+@TaskanaIntegrationTest
+class CompleteTaskWithSpiAccTest {
+
+  @TaskanaInject TaskService taskService;
+
+  ClassificationSummary defaultClassificationSummary;
+  WorkbasketSummary defaultWorkbasketSummary;
+  ObjectReference defaultObjectReference;
+
+  @SuppressWarnings("JUnitMalformedDeclaration")
+  @WithAccessId(user = "businessadmin")
+  @BeforeAll
+  void setup(ClassificationService classificationService, WorkbasketService workbasketService)
+      throws Exception {
+    defaultClassificationSummary =
+        defaultTestClassification().buildAndStoreAsSummary(classificationService);
+    defaultWorkbasketSummary = defaultTestWorkbasket().buildAndStoreAsSummary(workbasketService);
+
+    WorkbasketAccessItemBuilder.newWorkbasketAccessItem()
+        .workbasketId(defaultWorkbasketSummary.getId())
+        .accessId("user-1-1")
+        .permission(WorkbasketPermission.READ)
+        .permission(WorkbasketPermission.APPEND)
+        .buildAndStore(workbasketService);
+
+    defaultObjectReference = DefaultTestEntities.defaultTestObjectReference().build();
+  }
+
+  private TaskBuilder createTaskClaimedByUser(String user) {
+    return createDefaultTask().owner(user).state(TaskState.CLAIMED).claimed(Instant.now());
+  }
+
+  private TaskBuilder createDefaultTask() {
+    return TaskBuilder.newTask()
+        .classificationSummary(defaultClassificationSummary)
+        .workbasketSummary(defaultWorkbasketSummary)
+        .primaryObjRef(defaultObjectReference);
+  }
+
+  static class AlwaysRequireReview implements ReviewRequiredProvider {
+    @Override
+    public boolean reviewRequired(Task task) {
+      return true;
+    }
+  }
+
+  static class NeverRequireReview implements ReviewRequiredProvider {
+    @Override
+    public boolean reviewRequired(Task task) {
+      return false;
+    }
+  }
+
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  @WithServiceProvider(
+      serviceProviderInterface = ReviewRequiredProvider.class,
+      serviceProviders = AlwaysRequireReview.class)
+  class ServiceProviderAlwaysRequiringReview {
+
+    @TaskanaInject TaskService taskService;
+
+    @WithAccessId(user = "user-1-1")
+    @Test
+    void should_RequireReview_When_UserTriesToCompleteTask() throws Exception {
+      Task task = createTaskClaimedByUser("user-1-1").buildAndStore(taskService);
+
+      Task processedTask = taskService.completeTask(task.getId());
+
+      assertThat(processedTask.getState()).isEqualTo(TaskState.READY_FOR_REVIEW);
```

Comment:
Should we check that the owner is set to null as in requireReview?

Reply:
Since we've tested the functionality of requireReview intensively I didn't think that this is necessary.
All I wanted to verify in this test is that the Task is not completed because the SPI didn't allow it to.